common/include/pcl/conversions.h

```diff
@@ -91,7 +91,7 @@ namespace pcl
       template<typename Tag> void
       operator () ()
       {
-        BOOST_FOREACH (const pcl::PCLPointField& field, fields_)
+        for (const pcl::PCLPointField& field : fields_)
```

Comment:
Maybe `auto` since the type is already known (same in L214)

If you used sed/clang-tidy, you could modify it to replace the type too (`modernize-use-auto` or similar flag)

Reply:
See list of clang-tidy-checks [here](https://github.com/PointCloudLibrary/pcl/issues/2731) - we will not apply this modernization.